crates/rmf_site_editor/src/site/property_plugin.rs

```diff
@@ -188,15 +283,64 @@ fn on_add_property<T: Property, F: QueryFilter + 'static + Send + Sync>(
     T::insert(trigger.target(), scenario_entity, value.clone(), world);
 }
 
-fn on_add_root_scenario<T: Property, F: QueryFilter + 'static + Send + Sync>(
+/// When a new scenario has been created, this observer checks that it is a root
+/// scenario and calls T::insert_on_new_scenario so that the appropriate modifiers
+/// can be created for this Property via the callback.
+fn on_add_root_scenario<T: Property + 'static + Send + Sync, E: Element>(
     trigger: Trigger<OnAdd, ScenarioModifiers<Entity>>,
     world: &mut World,
-    state: &mut SystemState<Query<&Affiliation<Entity>, With<ScenarioMarker>>>,
+    state: &mut SystemState<Query<&Affiliation<Entity>>>,
+    events_state: &mut SystemState<EventWriter<ChangeCurrentScenario>>,
 ) {
     let scenarios = state.get_mut(world);
     if !scenarios.get(trigger.target()).is_ok_and(|p| p.0.is_none()) {
         return;
     }
 
-    T::insert_on_new_scenario(trigger.target(), world);
+    T::insert_on_new_scenario::<E>(trigger.target(), world);
```

Comment:
What if we generalize this a little bit, e.g.

```rust
T::on_new_scenario(trigger.target(), affiliation, world);
```

So `Property::on_new_scenario` is called for **_every_** new scenario and receives the `Affiliation` value for the scenario. Then the `Property::on_new_scenario` hook can be used for any new scenario, not just root scenarios, and implementations of the method can just check the affiliation value to see whether it's a root scenario or not.